errgo_handler.go

```diff
@@ -0,0 +1,52 @@
+package microerror
+
+import (
+	"fmt"
+
+	"github.com/juju/errgo"
+)
+
+// ErrgoHandler implements Handler interface.
+type ErrgoHandler struct {
+	callDepth int
+	maskFunc  func(err error, allow ...func(error) bool) error
+}
+
+func NewErrgoHandler() *ErrgoHandler {
+	return NewErrgoHandlerCallDepth(0)
+}
+
+// NewErrgoHandlerCallDepth is useful when creating a wrapper for ErrgoHandler.
+// The callDepth parameter is used to push stack location and skip wrapping
+// function location as an origin. The default value is 0.
+func NewErrgoHandlerCallDepth(callDepth int) *ErrgoHandler {
```

Comment:
I am not that big on it. I can always disagree and commit. I also do not say the stdlibs are wrong. I just think the way we do it with the default config approach is better and my inner monk fancies consistency a lot. :D